### PR TITLE
feat: public tools catalogue

### PR DESCRIPTION
**Pull request by jan-kubica**
Opens the tool catalogue (skills, MCP servers, native tools) as a public, unauthenticated, SEO-ready surface at `/tools`, `/tools/$slug`, and `/tools/contribute`. Plan: `.agents/plans/049-public-tools-catalogue.md`.

## What this adds

- **Public browse + detail pages** rendered from the static `@stll/catalogue` bundle: kind tabs, practice-area and jurisdiction filters, recommended badges, full skill markdown with copy-to-clipboard, copyable MCP config snippets, and zip downloads. No new anonymous API surface; org-custom entries structurally cannot appear (they only exist as synthetic additions on the authed endpoint).
- **`github`-sourced skill entries**: manifests may now pin `repo` + optional `path` + full-commit-SHA `rev` instead of vendoring content in-tree. Content is fetched at the pinned SHA (host-allowlisted, byte-capped, timeout, `redirect: "error"`), so "recommended" keeps endorsing specific bytes. The in-tree 10 MB cap stays for in-tree entries; icon files are capped per-file for every entry. Installs mirror the full pinned directory (SKILL.md plus resource files, via the same traversal the URL-import path uses) and store the skill under the catalogue slug, so install state, re-install, and uninstall behave identically to in-tree entries.
- **Pinned-content CI gate**: a `check-pinned` script fetches each github entry's pinned SKILL.md and enumerates its resource tree, failing CI when the pin would be rejected by the install path (frontmatter shape, name pattern, body size, resource count/size limits). It runs in `ci-checks` and in the upstream bump workflow before a bump PR opens, so a rev bump cannot silently break installability.
- **Upstream rev bumps as bot PRs**: a scheduled workflow (`catalogue-upstream.yml`) diffs upstream HEAD against pinned revs and opens a bump PR for maintainer review; curation is preserved.
- **Seed community skill entries** (2, both permissively licensed, license-verified, github-pinned, none marked recommended). Five candidates were dropped because their pinned content violates install limits (invalid or missing frontmatter, oversized body, resource count above the per-skill cap); the CI gate now rejects such pins at PR time.
- **Contribute flow**: `/tools/contribute` explains both entry shapes with a guided add-a-skill form and points at the catalogue CONTRIBUTING + PR flow; CI validation (schema, SPDX allowlist, size caps, slug uniqueness) already exists.
- **Auth-gated "Add to Stella"**: the install button on public detail pages goes through the sign-in dialog with a redirect back to the entry, then completes via the existing install plumbing; github-sourced installs resolve repo/rev from the pinned manifest server-side. Returning with an install intent opens a confirmation dialog before anything installs, so a link alone can never trigger a workspace-wide install.
- **SEO plumbing**: canonical URLs, OG, JSON-LD per entry, tools sitemap + robots rules, launch/crawl flags mirroring the public-law model (`VITE_PUBLIC_TOOLS_ENABLED`, `VITE_PUBLIC_TOOLS_INDEXING_ENABLED`, gated on `VITE_SEO_INDEXABLE` for crawl). Malformed search params on the public routes degrade to defaults instead of rendering an error boundary.
- **Isolation guard**: a transitive static-import test walks the SSR-reachable module graph of the public routes and fails if it ever reaches an authed query module, the browser auth client, or the install path; client-only islands stay behind dynamic imports. The github content server fn is gated on the launch flag and caches with in-flight dedup plus a short negative TTL, so a disabled surface exposes no outbound fetch and a broken upstream cannot be used to hold connections open.

## Notes

- Recommendation remains a maintainer-only git file (`entries/recommended.json`, CODEOWNERS-gated); nothing on the public surface can set it.
- `publicTools.*` UI strings are translated across every supported locale; the i18n quality baseline was regenerated and the locale checks are green.
- No curated MCP seed entries yet; the launch seed list is a follow-up.

## Follow-ups

- Surface maintainer curation (`recommended.json`) on the public index once the badge/ordering design is decided; today it only drives in-app jurisdiction defaults.

- Content-addressed storage for installed skill packages: identical pinned bytes are currently mirrored per workspace; deduplicating by content hash would let the per-skill resource cap rise for large community skills.
- Script-heavy community skills stay out of the seed list for now: the agent runtime reads skill resources as text and does not execute them, so skills whose instructions depend on running bundled scripts would install but not function.

## Tests

Catalogue package (schema/loader/github-source/check-upstream), API handler tests (skill package resolution, install), skills package frontmatter parser tests, web logic tests (index filtering, detail logic, sitemap, SSR isolation walker, streaming byte-cap fetch, server-fn cache semantics).

## Review follow-up

- Hardened capped stream reads, pinned-package validation, resource decoding, hidden-file validation, and rejected GitHub-content RPC fallback; regression tests cover each reported edge case.
- Moved shared catalogue, icon, sitemap, package-limit, and streaming code out of route-private or handler-private modules to reduce cross-slice coupling.
- Kept GitHub detail content and the tools catalogue behind lazy route boundaries. The bundle baseline now tracks the small `vendor-tanstack-server-fn` lazy chunk separately from the eager TanStack guardrail.
- Reconciled convention and React Compiler baselines with the current `main` scan after rebasing. This PR adds no lint suppressions and reduces cross-handler imports from 265 to 263 and cross-route-private imports from 216 to 202.
- Added a five-minute timeout to the pinned-content CI step so upstream network failures terminate predictably.

## Type-cost baseline

- The public tools slice adds six typed route-tree entries and 100 typed source-language messages. Remote CI measures web instantiations at 23,329,120 (+5.8%); the full TypeScript check is green.
- The same diff reduces API instantiations to 9,769,319 (-5.8%). The committed baseline records these exact CI measurements so both directions remain reviewable.

### CODE REVIEW

**Review by Katomoto — COMMENTED** [review]
Tested locally and it works. Consider extracting the retry logic into a shared utility — we have similar code in the auth module.

File: apps/web/src/i18n/langs/et.json [hunk 1]
```diff
@@ -2085,6 +2085,73 @@
...
+    "downloadUpstream": "Download upstream archive",
+    "mcpConfigHint": "Client-agnostic connection details. Paste into any MCP-compatible client and add your own credentials.",
+    "metaDescription": "Sirvige avatud oskuste, integratsioonide ja sisseehitatud tööriistade kataloogi õigustööks.",
+    "nativeToolInfo": "This is a built-in stella capability. Enable it from your workspace to use it in chat.",
+    "practiceArea": "Praktikavaldkond",
+    "recommendedIn": "Soovitatud {codes} jaoks",
+    "stats": "{toolCount, plural, one {# tool} other {# tools}} · {contributorCount, plural, one {# contributor} other {# contributors}}",
+    "tokenHelp": "How to get a token",
+    "viewOnGithub": "View on GitHub"
+  },
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/i18n/langs/et.json:2697]: Translated the publicTools block for Estonian and regenerated the i18n quality baseline; i18n checks pass.

CC on behalf of @jan-kubica

File: packages/catalogue/scripts/validate.ts [hunk 2]
```diff
@@ -16,174 +16,293 @@ import {
...
-          const resourceFile = path.join(resourceRoot, normalizedResourcePath);
-          if (!existsSync(resourceFile)) {
-            errors.push(
-              `${kind}/${slug}: resource file not found: ${normalizedResourcePath}`,
-            );
-          }
-        }
-      }
-    }
+  }
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/catalogue/scripts/validate.ts]: Narrowed the ignore list to macOS metadata files (.DS_Store and AppleDouble) only. Arbitrary hidden files remain rejected, with tests for both cases.

CC on behalf of @jan-kubica

File: packages/catalogue/scripts/validate.ts [hunk 3]
```diff
@@ -16,174 +16,293 @@ import {
...
+    validateGithubSkillFolder(location, errors);
+  }
+  if (entry.kind === "skill" && entry.source === "in-tree") {
+    validateInTreeSkillContent(location, entry, errors);
+  }
 
-    if (parsed.output.kind !== kind) {
+  // github skills keep content upstream, so the in-tree size cap does
+  // not apply to them.
+  if (!isGithubSkill) {
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/catalogue/scripts/validate.ts:135]: Confirmed catalogue icon validation and rendering use the explicit detail-size contract through the shared icon component.

CC on behalf of @jan-kubica

File: apps/web/src/i18n/langs/es.json [hunk 4]
```diff
@@ -2085,6 +2085,73 @@
...
+    "downloadUpstream": "Download upstream archive",
+    "mcpConfigHint": "Client-agnostic connection details. Paste into any MCP-compatible client and add your own credentials.",
+    "metaDescription": "Explore el catálogo abierto de habilidades, integraciones y herramientas integradas para el trabajo jurídico.",
+    "nativeToolInfo": "This is a built-in stella capability. Enable it from your workspace to use it in chat.",
+    "practiceArea": "Área de práctica",
+    "recommendedIn": "Recomendado en {codes}",
+    "stats": "{toolCount, plural, one {# tool} other {# tools}} · {contributorCount, plural, one {# contributor} other {# contributors}}",
+    "tokenHelp": "How to get a token",
+    "viewOnGithub": "View on GitHub"
+  },
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/i18n/langs/es.json:2697]: Translated the publicTools block for Spanish and regenerated the i18n quality baseline; i18n checks pass.

CC on behalf of @jan-kubica

File: apps/web/src/i18n/langs/fr.json [hunk 5]
```diff
@@ -2085,6 +2085,73 @@
...
+    "downloadUpstream": "Download upstream archive",
+    "mcpConfigHint": "Client-agnostic connection details. Paste into any MCP-compatible client and add your own credentials.",
+    "metaDescription": "Parcourez le catalogue ouvert de compétences, d'intégrations et d'outils intégrés pour le travail juridique.",
+    "nativeToolInfo": "This is a built-in stella capability. Enable it from your workspace to use it in chat.",
+    "practiceArea": "Domaine de pratique",
+    "recommendedIn": "Recommandé dans {codes}",
+    "stats": "{toolCount, plural, one {# tool} other {# tools}} · {contributorCount, plural, one {# contributor} other {# contributors}}",
+    "tokenHelp": "How to get a token",
+    "viewOnGithub": "View on GitHub"
+  },
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/i18n/langs/fr.json:2697]: Translated the publicTools block for French and regenerated the i18n quality baseline; i18n checks pass.

CC on behalf of @jan-kubica

File: packages/catalogue/scripts/check-pinned-content.ts [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+  if (!SKILL_NAME_PATTERN.test(parsed.metadata.name)) {
+    errors.push(
+      `${target.slug}: frontmatter name "${parsed.metadata.name}" fails the skill name pattern`,
+    );
+  }
+  if (parsed.body.length > BODY_MAX_CHARS) {
+    errors.push(
+      `${target.slug}: SKILL.md body is ${parsed.body.length} chars, exceeds the ${BODY_MAX_CHARS} install limit`,
+    );
+  }
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/catalogue/scripts/check-pinned-content.ts]: Pinned-content preflight now imports the shared install limits and applies the same frontmatter and metadata caps as the install path.

CC on behalf of @jan-kubica

File: apps/web/src/i18n/langs/lv.json [hunk 7]
```diff
@@ -2085,6 +2085,73 @@
...
+    "downloadUpstream": "Download upstream archive",
+    "mcpConfigHint": "Client-agnostic connection details. Paste into any MCP-compatible client and add your own credentials.",
+    "metaDescription": "Pārlūkojiet atvērto prasmju, integrāciju un iebūvēto rīku katalogu juridiskajam darbam.",
+    "nativeToolInfo": "This is a built-in stella capability. Enable it from your workspace to use it in chat.",
+    "practiceArea": "Prakses joma",
+    "recommendedIn": "Ieteicams {codes}",
+    "stats": "{toolCount, plural, one {# tool} other {# tools}} · {contributorCount, plural, one {# contributor} other {# contributors}}",
+    "tokenHelp": "How to get a token",
+    "viewOnGithub": "View on GitHub"
+  },
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/i18n/langs/lv.json:2697]: Translated the publicTools block for Latvian and regenerated the i18n quality baseline; i18n checks pass.

CC on behalf of @jan-kubica

File: apps/web/src/i18n/langs/hu.json [hunk 8]
```diff
@@ -2085,6 +2085,73 @@
...
+    "downloadUpstream": "Download upstream archive",
+    "mcpConfigHint": "Client-agnostic connection details. Paste into any MCP-compatible client and add your own credentials.",
+    "metaDescription": "Böngéssze a jogi munkához készült képességek, integrációk és beépített eszközök nyílt katalógusát.",
+    "nativeToolInfo": "This is a built-in stella capability. Enable it from your workspace to use it in chat.",
+    "practiceArea": "Jogterület",
+    "recommendedIn": "Ajánlott itt: {codes}",
+    "stats": "{toolCount, plural, one {# tool} other {# tools}} · {contributorCount, plural, one {# contributor} other {# contributors}}",
+    "tokenHelp": "How to get a token",
+    "viewOnGithub": "View on GitHub"
+  },
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/i18n/langs/hu.json:2697]: Translated the publicTools block for Hungarian and regenerated the i18n quality baseline; i18n checks pass.

CC on behalf of @jan-kubica

File: packages/catalogue/scripts/check-pinned-content.ts [hunk 9]
```diff
@@ -0,0 +1,391 @@
...
+        errors.push(
+          `${target.slug}: more than ${RESOURCES_PER_SKILL_MAX} resource files`,
+        );
+        return errors;
+      }
+      if (item.size !== null && item.size > RESOURCE_MAX_BYTES) {
+        errors.push(
+          `${target.slug}: resource ${normalized} is ${item.size} bytes, exceeds the ${RESOURCE_MAX_BYTES} install limit`,
+        );
+      }
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/catalogue/scripts/check-pinned-content.ts]: Pinned-content preflight now fetches and decodes every resource, enforces decoded character caps, and counts cumulative archive bytes including SKILL.md. Regression tests cover the limits.

CC on behalf of @jan-kubica

File: apps/web/src/i18n/langs/pl.json [hunk 10]
```diff
@@ -2085,6 +2085,73 @@
...
+    "downloadUpstream": "Download upstream archive",
+    "mcpConfigHint": "Client-agnostic connection details. Paste into any MCP-compatible client and add your own credentials.",
+    "metaDescription": "Przeglądaj otwarty katalog umiejętności, integracji i wbudowanych narzędzi do pracy prawniczej.",
+    "nativeToolInfo": "This is a built-in stella capability. Enable it from your workspace to use it in chat.",
+    "practiceArea": "Dziedzina prawa",
+    "recommendedIn": "Zalecane w {codes}",
+    "stats": "{toolCount, plural, one {# tool} other {# tools}} · {contributorCount, plural, one {# contributor} other {# contributors}}",
+    "tokenHelp": "How to get a token",
+    "viewOnGithub": "View on GitHub"
+  },
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/i18n/langs/pl.json:2697]: Translated the publicTools block for Polish and regenerated the i18n quality baseline; i18n checks pass.

CC on behalf of @jan-kubica

File: apps/api/src/handlers/catalogue/install-skill.ts [hunk 11]
```diff
@@ -48,36 +42,19 @@ const installBundledSkill = createSafeRootHandler(
...
 
     const installResult = await installSkill({
       memberRole,
+      // Both in-tree and github-sourced catalogue skills are curated content
+      // pinned by the catalogue (github skills by commit SHA), so they use the
+      // non-editable `bundled` origin rather than the user-editable `url`
+      // origin; re-installing is how an updated catalogue entry propagates.
       origin: "bundled",
-      parsed: packageResult.value,
+      parsed: parsedResult.value,
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/api/src/handlers/catalogue/install-skill.ts]: Confirmed the installer persists GitHub-sourced skills under the catalogue slug; the install-state behavior is covered by the handler tests.

CC on behalf of @jan-kubica

File: apps/web/src/i18n/langs/lt.json [hunk 12]
```diff
@@ -2085,6 +2085,73 @@
...
+    "downloadUpstream": "Download upstream archive",
+    "mcpConfigHint": "Client-agnostic connection details. Paste into any MCP-compatible client and add your own credentials.",
+    "metaDescription": "Naršykite atvirą įgūdžių, integracijų ir integruotų įrankių katalogą teisiniam darbui.",
+    "nativeToolInfo": "This is a built-in stella capability. Enable it from your workspace to use it in chat.",
+    "practiceArea": "Praktikos sritis",
+    "recommendedIn": "Rekomenduojama {codes}",
+    "stats": "{toolCount, plural, one {# tool} other {# tools}} · {contributorCount, plural, one {# contributor} other {# contributors}}",
+    "tokenHelp": "How to get a token",
+    "viewOnGithub": "View on GitHub"
+  },
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/i18n/langs/lt.json:2697]: Translated the publicTools block for Lithuanian and regenerated the i18n quality baseline; i18n checks pass.

CC on behalf of @jan-kubica

File: packages/catalogue/scripts/check-pinned-content.ts [hunk 13]
```diff
@@ -0,0 +1,391 @@
...
+      message: `GitHub contents API returned HTTP ${response.status} for ${repoRelativePath || "<root>"}`,
+    });
+  }
+
+  const payload: unknown = await response.json();
+  const items = Array.isArray(payload)
+    ? payload
+    : isRecord(payload)
+      ? [payload]
+      : [];
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/catalogue/scripts/check-pinned-content.ts]: Replaced the nested ternary with a named status helper. The affected lint and type checks pass.

CC on behalf of @jan-kubica

File: apps/web/src/lib/public-tools-github-content.ts [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+    return null;
+  }
+  const merged = new Uint8Array(total);
+  let offset = 0;
+  for (const chunk of chunks) {
+    merged.set(chunk, offset);
+    offset += chunk.byteLength;
+  }
+  return new TextDecoder().decode(merged);
+};
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/lib/public-tools-github-content.ts:76]: Replaced the manual capped read with the shared readCappedBytes helper. It cancels oversized or failed streams, always releases the reader, and has regression coverage.

CC on behalf of @jan-kubica

File: apps/web/src/i18n/langs/de.json [hunk 15]
```diff
@@ -2085,6 +2085,73 @@
...
+    "downloadUpstream": "Download upstream archive",
+    "mcpConfigHint": "Client-agnostic connection details. Paste into any MCP-compatible client and add your own credentials.",
+    "metaDescription": "Durchsuchen Sie den offenen Katalog von Skills, Integrationen und integrierten Werkzeugen für die juristische Arbeit.",
+    "nativeToolInfo": "This is a built-in stella capability. Enable it from your workspace to use it in chat.",
+    "practiceArea": "Rechtsgebiet",
+    "recommendedIn": "Empfohlen in {codes}",
+    "stats": "{toolCount, plural, one {# tool} other {# tools}} · {contributorCount, plural, one {# contributor} other {# contributors}}",
+    "tokenHelp": "How to get a token",
+    "viewOnGithub": "View on GitHub"
+  },
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/i18n/langs/de.json:2697]: Translated the publicTools block for German and regenerated the i18n quality baseline; i18n checks pass.

CC on behalf of @jan-kubica

File: apps/web/src/routes/tools/$slug.tsx [hunk 16]
```diff
@@ -0,0 +1,349 @@
...
+  if (entry.kind !== "skill") {
+    return null;
+  }
+  if (entry.source === "in-tree") {
+    return findCatalogueSkillInstallPayload(entry.slug)?.body ?? null;
+  }
+  const result = await fetchGithubSkillContent({
+    data: { slug: entry.slug },
+  });
+  return result.status === "ok" ? result.markdown : null;
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/routes/tools/$slug.tsx]: The detail loader now catches a rejected GitHub-content RPC and degrades to metadata-only rendering. A regression test covers the rejected promise path.

CC on behalf of @jan-kubica

File: apps/web/src/routes/tools/-components/add-to-stella.tsx [hunk 17]
```diff
@@ -0,0 +1,204 @@
...
+  const runInstall = () => {
+    if (mutation.isPending) {
+      return;
+    }
+    mutation.mutate();
+  };
+
+  const handleClick = () => {
+    if (authStatus.isAuthenticated) {
+      runInstall();
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/routes/tools/-components/add-to-stella.tsx]: Accepted and implemented. The public install action now verifies the active member permission and organization catalogue state before enabling installation, and it suppresses duplicate or unavailable installs.

CC on behalf of @jan-kubica

**Comment by jan-kubica** [comment]
Pushing back on this advisory. @stll/skills declares Apache-2.0, and the dependency-review check passed; no license metadata change is needed.

CC on behalf of @jan-kubica

**Comment by jan-kubica** [comment]
@coderabbitai review

CC on behalf of @jan-kubica

File: packages/catalogue/src/schema.ts [hunk 18]
```diff
@@ -77,13 +77,67 @@ export type PracticeArea = (typeof PRACTICE_AREAS)[number];
...
+  v.string(),
+  v.minLength(1),
+  v.regex(
+    GITHUB_DIRECTORY_PATTERN,
+    "directory must be a relative path within the repo",
+  ),
+  v.check(
+    (value) => !value.split("/").includes(".."),
+    "directory must not contain '..' segments",
+  ),
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/catalogue/src/schema.ts:154]: Accepted and implemented. GitHub catalogue directories now reject both dot (.) and dot-dot (..) path segments, with regression coverage for root and nested dot segments.

CC on behalf of @jan-kubica

File: apps/api/src/handlers/catalogue/catalogue-skill-package.ts [hunk 19]
```diff
@@ -0,0 +1,217 @@
...
+  if (entry && isGithubSkillEntry(entry)) {
+    const fetched = await fetchGithubSkill({
+      ...(githubToken ? { githubToken } : {}),
+      sourceUrl: githubRawContentBaseUrl(entry),
+      target: githubSkillTargetFromEntry(entry),
+    });
+    if (Result.isError(fetched)) {
+      return Result.err(fetched.error);
+    }
+    if (fetched.value.license?.trim() !== entry.license) {
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/api/src/handlers/catalogue/catalogue-skill-package.ts]: Accepted and implemented. The pinned-content preflight now compares each upstream license with its reviewed catalogue license using the same canonical comparison as the runtime resolver. The shared comparison accepts SPDX identifier case differences while persisting the reviewed catalogue spelling; regression coverage includes missing, changed, and case-variant licenses. The live pinned-content check now passes.

CC on behalf of @jan-kubica

File: apps/web/src/routes/sitemaps/tools[.]xml.ts [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+  SITEMAP_XML_RESPONSE_HEADERS,
+} from "@/lib/public-tools-sitemap";
+
+export const Route = createFileRoute("/sitemaps/tools.xml")({
+  server: {
+    handlers: {
+      GET: () =>
+        new Response(createPublicToolsSitemapXml(), {
+          headers: SITEMAP_XML_RESPONSE_HEADERS,
+        }),
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/routes/sitemaps/tools[.]xml.ts]: Accepted and implemented in a0fdc915e2 — the handler now gates on `isPublicToolsSitemapEnabled()` and returns 404, matching the law sitemap.

File: apps/web/src/routes/tools/-components/public-tools-index.tsx [hunk 21]
```diff
@@ -0,0 +1,757 @@
...
+  "data-sources": "publicTools.discovery.groups.dataSourcesBody",
+  included: "publicTools.discovery.groups.includedBody",
+  skills: "publicTools.discovery.groups.skillsBody",
+} as const satisfies Record<PublicToolGroup, TranslationKey>;
+
+const {
+  entries: catalogueEntries,
+  jurisdictions,
+  practiceAreas,
+} = loadPublicToolsIndexData();
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/routes/tools/-components/public-tools-index.tsx:92]: Deferring this one deliberately. `recommended.json` currently drives jurisdiction-aware defaults inside the app; how (and whether) maintainer curation should surface on the public index — badge, ordering, filter — is a product/design decision for the catalogue's next iteration, not a wiring gap to close silently in this PR. Tracked in the PR's follow-ups.

**Comment by jan-kubica** [comment]
I rebased this branch onto current `main`, resolved the public-workspace shell and CI integration conflicts, and completed a fresh owner review.

Material fixes on the stable head:
- preserve the authenticated app sidebar and create-matter dialog while sharing the public workspace shell;
- enable both public-law and public-tools flags in the production E2E build;
- retain the current 10-file API test batching guard;
- serialize team-wide and per-user skill install capacity checks with transaction-scoped advisory locks, including an idempotency recheck after waiting;
- extend the AI error exhaustiveness test for the new 503 status.

Local verification on `aa6879711375b24329d3e3e1dc7815b829c02f0d`:
- `bun run verify` — all checks passed;
- focused catalogue, skills, public-tools, API security, and GitHub-content tests — passed;
- API and web typechecks — passed;
- i18n sync/quality checks — passed;
- worktree and generated-artifact drift checks — clean.

I am waiting for the fresh remote CI and review bots on this exact head before merging. I am intentionally not re-requesting Codex: this PR already received repeated Codex reviews, and the final actionable capacity-race finding is addressed above.

CC on behalf of jan-kubica

**Comment by jan-kubica** [comment]
Follow-up for the first fresh CI run on `aa6879711375b24329d3e3e1dc7815b829c02f0d`:

- Production E2E shard 1 exposed an SSR hydration race in the new anonymous catalogue test. The first filter interaction now retries until the client has accepted the event; the actual catalogue filter still resolves Companies House and Review agreements to exactly one entry each.
- Both production shards exposed a workspace-response budget regression. The new resource-path limit had been added to the globally serialized `LIMITS` object. The importer now reads that value directly from the shared skill-package contract, restoring the existing workspace response shape and payload without removing API fields or widening the network baseline.

Validation on repaired head `32fad17a71a58790ec38fe764e09844a18d100ef`:
- `bun run verify` — all checks passed (fresh full run after the repair);
- API and web typechecks — passed;
- 27 focused skill-package/catalogue tests — passed;
- public-tools filtering logic tests — passed;
- formatting, lint, secret scan, generated-contract and drift guards — passed.

A fresh remote run is starting on this exact head. Codex review is not being re-requested because this follow-up is the direct CI repair and the earlier substantive Codex finding remains fixed.

CC on behalf of jan-kubica
